src/Microsoft.AspNetCore.Authentication.OpenIdConnect/OpenIdConnectHandler.cs

```diff
@@ -350,27 +349,38 @@ public OpenIdConnectHandler(HttpClient backchannel, HtmlEncoder htmlEncoder)
 
             try
             {
-                var messageReceivedContext = await RunMessageReceivedEventAsync(authorizationResponse);
-                if (CheckEventResult(messageReceivedContext, out result))
+                AuthenticationProperties properties = null;
+                if (!string.IsNullOrEmpty(authorizationResponse.State))
+                {
+                    properties = Options.StateDataFormat.Unprotect(Uri.UnescapeDataString(authorizationResponse.State));
```

Comment:
Is `Uri.UnescapeDataString` really mandatory? I'm pretty sure the OAuth2 MW doesn't use it.